tests/integration/models/device.spec.coffee

```diff
@@ -1369,3 +1450,235 @@ describe 'Device Model', ->
 				m.chai.expect(
 					balena.models.device.lastOnline(mockDevice)
 				).to.equal('5 minutes ago')
+
+		describe 'balena.models.device._checkOsUpdateTarget()', ->
+
+			uuid = balena.models.device.generateUniqueKey()
+
+			it 'should throw when the current os version is invalid', ->
+				[
+					['Resin OS ', 'dev']
+					['Resin OS ', 'prod']
+					['Resin OS 2.0-beta.8', '']
+				].forEach ([os_version, os_variant]) ->
+					mockDevice = {
+						uuid
+						device_type: 'raspberrypi3'
+						is_online: true
+						os_version
+						os_variant
+					}
+
+					m.chai.expect(-> balena.models.device._checkOsUpdateTarget(
+						mockDevice
+						'2.29.2+rev1.prod'
+					)).to.throw('Invalid current balenaOS version')
+
+			it 'should throw when the current os version is a very old v1', ->
```

Comment:
Is it because it's "old", or is it because we exclude devices that are OS version < 1.8.0, as they had a file system fingerprinting bug, that would prevent proper updates? :)  (that's the reason it's not self serve for those versions)

Reply:
`< 1.8.0` is what we are actually checking, but I though to not make it part of the rules, since this is already checked in the hup utils module. Now that I'm thinking it again, let's change it, so that we remember how/why the test case was written.